nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -393,7 +393,13 @@ in
             };
             script = ''
               zpool_cmd="${packages.zfsUser}/sbin/zpool"
-              ("$zpool_cmd" list "${pool}" >/dev/null) || "$zpool_cmd" import -d ${cfgZfs.devNodes} -N ${optionalString cfgZfs.forceImportAll "-f"} "${pool}"
+              echo -n "importing ZFS pool \"${pool}\"..."
+              "$zpool_cmd" list "${pool}" >/dev/null 2>/dev/null && exit
+              for trial in `seq 1 60`; do
```

Comment:
There should be a comment here, why this is a loop.

Reply:
Fixed.